kstatus/README.md

```diff
@@ -0,0 +1,116 @@
+that the library will understand and that we encourage developers to adopt in their CRDs. These standard conditions will
+be focused on providing the necessary information for understanding status of the reconcile after `apply` and it is not
+expected that these will necessarily be the only conditions exposed by a CRD. Developers will be free to add as many conditions
+as they wish, but if the CRDs expose the standard conditions defined here, this library will handle them correctly.
+
+The `status` objects for built-in types are currently quite different. Not all built-in types expose conditions,
+and even among the types that does, the types of conditions vary widely. Long-term, we hope to add support for the
+standard conditions into the built-in types as well. This would remove the need for type-specific rules for determining
+status.
+
+## Statuses
+
+The library currently defines the following statuses for resource:
+* __InProgress__: The actual state of the resource has not yet reached the desired state as specified in the
+resource manifest, i.e. the resource reconcile has not yet completed.
+* __Failed__: The process of reconciling the actual state with the desired state has encountered and error
+or it has made insufficient progress.
+* __Current__: The actual state of the resource matches the desired state. The reconcile process is considered
+complete until there are changes to either the desired or the actual state.
+* __Terminating__: The resource is in the process of being deleted.
+* __Unknown__: This is for situations when the library are unable to determine the status of a resource.
+
+## Conditions
+
+The conditions defined in the library are designed to adhere to the "abnormal-true" pattern, i.e. that 
+conditions are present and with a status of true whenever something unusual happens. So the absence of
+any conditions means everything is normal. Normal in this situation simply means that the latest observed 
+generation of the resource manifest by the controller have been fully reconciled with the actual state. 
+
+* __InProgress__: The controller is currently working on reconciling the latest changes.
+* __Failed__: The controller has encountered an error during the reconcile process or it has made
+insufficient progress (timeout).
+
+The use of the "abnormal-true" pattern has some challenges. If the controller is not running, or for some
+reason not able to update the resource, it will look like it is in a good state when that is not true. The
+solution to this issue is to adopt the pattern used by several of the built-in types where there is an
+`observedGeneration` property on the status object which is set by the controller during the reconcile loop.
+If the `generation` and the `observedGeneration` of a resource does not match, it means there are changes
+that the controller has not yet seen, and therefore not acted upon.
```

Comment:
Same question.  We adding more conditions or hiding them?

Reply:
I think similar to status, these are conditions focused on answering questions about the progress/failures of the reconcile process. There are probably use cases that will require other conditions.